examples/multigrain-simulation.ipynb

```diff
@@ -0,0 +1,439 @@
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "233f83f8-4d10-4c4e-acbb-7b4ac07616ab",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<astropy.units.core._UnitContext at 0x79402b9a6790>"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "u.set_enabled_equivalencies(u.magnetic_flux_field())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "559d9402-1d82-4330-acda-278d4276a9cd",
+   "metadata": {},
+   "source": [
+    "## The multigrain material\n",
+    "The mesh `cube40_equi_grains8_gsize20` describe a cube of side length 40 containing 8 different grains generated from a uniform grain distribution with the parameter grain size set to 20.\n",
+    "The space between the different grain represents the boundary interface.\n",
+    "The cube is surrounded by a spherical non-magnetic material further surrounded by a shell.\n",
+    "\n",
+    "Hence the total number of regions is 11:\n",
+    "- 8 grains\n",
+    "- 1 boundary interface\n",
+    "- 1 non-magnetic material\n",
+    "- 1 shell."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "49429b0a-64a8-497a-b139-e0549a3c4ab1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mesh = \"cube40_equi_grains8_gsize20\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0c4ceb92-67b3-4dab-bba9-284398d5639c",
+   "metadata": {},
+   "source": [
+    "## Simulation using the `hysteresis.run` function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "71fabe3e-f9de-434a-92ba-bee239853fc2",
+   "metadata": {},
+   "source": [
+    "We can assume the intrinsic properties in all grains have the same value (the generalization is trivial):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "0d9029e0-67ac-4b12-b90b-bf1454c8d9a1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Ms = me.Ms([1280000] * 8)\n",
+    "A = me.A([7.7e-12] * 8)\n",
+    "Ku = me.Ku([4300000] * 8)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "353b9fad-4694-4a35-b788-be78d70a18a2",
+   "metadata": {},
+   "source": [
+    "While the `hysteresis.run` function always assume the magnetic parameters to be zero in the non-magnetic material and in the shell, we still need to assign a value to the boundary interface.\n",
+    "In this case we assign zero.\n",
+    "\n",
+    "To concatenate a number to an entity, we use the `concat_flat` function from `mammos_entity`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "382ac95d-6386-48a2-bce4-335bbe685a8a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hysteresis_result = hysteresis.run(\n",
+    "    mesh=mesh,\n",
+    "    Ms=me.concat_flat(Ms, 0),\n",
+    "    A=me.concat_flat(A, 0),\n",
+    "    K1=me.concat_flat(Ku, 0),\n",
+    "    theta=[0.0] * 9,\n",
+    "    phi=[0.0] * 9,\n",
```

Comment:
Should we at-least change the direction of anisotropy for each grain to get some interesting result? Initial thought: Keeping `phi` constant, pick 8 random values of `theta` between 0 and $\frac{\pi}{2}$?